### PR TITLE
[TC-532] add tenancy check to ds safe-update endpoint

### PR DESCRIPTION
**Pull request by DylanVolz**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/142/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/103/

File: traffic_ops/app/lib/API/Deliveryservice.pm [hunk 1]
```diff
@@ -480,6 +480,12 @@ sub safe_update {
 		return $self->not_found();
 	}
 
+	my $tenant_utils = Utils::Tenant->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db();
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id)) {
```
Comment (mitchell852) [inline traffic_ops/app/lib/API/Deliveryservice.pm]: this is not quite right. i think the test needs to be 

if (use_tenancy() and !$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id))

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-whatchanged/3/

File: traffic_ops/app/lib/API/Deliveryservice.pm (hunk 1 above)
Comment (mitchell852) [inline traffic_ops/app/lib/API/Deliveryservice.pm]: nm, use_tenancy is checked inside is_ds_resource_accessible

**Pull request merged**